_pages/dev/ssr/server-side-rendering-error-handling.md

```diff
@@ -0,0 +1,349 @@
+- `ErrorAction` - an interface with a property `error`, used in `CxErrorHandlerEffect` to filter error actions.
+
+The following is an example of how to implement the `ErrorAction` interface in your custom failure action: 
+```ts
+export class MyActionFail implements ErrorAction {
+  readonly type = MY_ACTION_FAIL;
+  public error: Object;
+  constructor(public payload: Object) {
+    this.error = payload;
+  }
+}
+```
+Please make sure that all your custom NgRx Actions of type Fail contain the property `error`.
+
+**Note:** If in your case some exceptional NgRx actions with the property `error` should not make SSR fail, you can customize the behavior by over-providing Spartacus `ErrorActionService`.
+
+## Don't use the RESPONSE token for setting status code
+
+Before the provided solutions, one of the way to react on handled errors was to manually inject `RESPONSE` token in their code and call `this.response.set(<custom-status>)`. Customers who have so far used this approach should no longer do so as it's not guaranteed to work correctly with the new contract between Angular app and ExpressJS. If there is a need for handling any custom error, it can be done by injecting `ErrorHandler` and calling `this.errorHandler.handleError(<custom-error>)` instead.
+
+```ts
+@Injectable()
+export class MyService {
+  const errorHandler = inject(ErrorHandler);
+
+  myMethod() {
+    try {
+      // some code that may throw an error
+    } catch (error) {
+      this.errorHandler.handleError(error);
+    }
+  }
+}
+```
+
+## Cache management and error handling
+
+It is not recommended to cache pages for which an error occurred during rendering due to several reasons:
+- *Serving Incorrect Content:* If a page with rendering errors is cached, subsequent requests to that page will receive the erroneous content. This can lead to a poor user experience and confusion, as users might see incomplete or broken pages.
+
+- *Difficulty in Debugging:* When errors are cached, it becomes harder to identify and debug issues. Developers might not be aware that users are receiving erroneous content, making it challenging to diagnose and fix the underlying problem.
+
+- *Performance Degradation:* While caching is generally used to improve performance, caching pages with errors can have the opposite effect. Users might repeatedly encounter the same error, leading to frustration and potentially increasing the load on support channels.
+
+To avoid caching such pages, Spartacus provides changes to the part of `OptimizedSsrEngine`, the `RenderingCache` class, responsible for caching renders. From now, the caching strategy can be controlled by the new `SsrOptimizationOptions` property, `shouldCacheRenderingResult`:
+```ts
+shouldCacheRenderingResult?: ({
+    options,
+    entry,
+  }: {
+    options: SsrOptimizationOptions;
+    entry: Pick<RenderingEntry, 'err' | 'html'>;
+  }) => boolean;
+```
```

Comment:
Let's also document the new option in the main list of all the `SsrOptimizationOptions` in the official docs:
https://github.com/SAP/spartacus-docs/blob/develop/_pages/dev/ssr/server-side-rendering-optimization.md